### PR TITLE
Use remote Kustomize base for deploying AKS Periscope

### PR DESCRIPTION
**Pull request by peterbom**
The latest Periscope release provides a Kustomize base file for deploying Periscope with Windows support.

These changes make use of Kustomize to remove the need for magic string replacement, and minimize the required changes to az-cli when new Periscope versions are released (we'll only have to update the release version and image tag, not the entire deployment yaml).

NOTE: the minimum supported kubectl version for Kustomize is [1.14](https://github.com/kubernetes/sig-release/blob/master/releases/release-1.14/release-notes-draft.md) (released March 2019).

cc. @Tatsinnit, @FumingZhang 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Review by Tatsinnit — APPROVED** [review]
☕️💡 LGTM , as long as it’s tested, which is the case 🙏❤️

**Review by FumingZhang — APPROVED** [review]
LGTM

**Comment by zhoxing-ms** [comment]
Could you add some test for these change?

**Comment by zhoxing-ms** [comment]
Please add the description in history notes `HISTORY.rst`. If you need to upgrade the extension version to release, please also modify `setup.py`

**Comment by peterbom** [comment]
> Could you add some tests for these change?

I'm attempting to add a test now, but running into the issue that the command is interactive (there are a couple of prompts: one to confirm the user consents to the operation, and one to find out whether the user wishes to see the analysis results).

The common approach to avoid this seems to be to add a `--yes` ("do not prompt for confirmation") parameter, but:
- So far I can't quite understand the code required to do this (like why some parameters are defined in `_params.py` and some in `_help.py`.
- This particular command has _two_ different prompts, and it's unclear what the default should be for the second.

All in all, this change seems to entail a non-trivial bit of design work that's not relevant to the intent of the original PR.

Should we address this (and add the test) in a separate PR?

@zhoxing-ms, @FumingZhang, @Tatsinnit for thoughts.

**Comment by zhoxing-ms** [comment]
> All in all, this change seems to entail a non-trivial bit of design work that's not relevant to the intent of the original PR.
> Should we address this (and add the test) in a separate PR?

LGTM

**Comment by FumingZhang** [comment]
Yeah, would appreciate if you could add some unit test for these commands, since they are not related to ARM resource, I suppose live/e2e test based on vcr/recording is not suitable.

File: src/aks-preview/azext_aks_preview/custom.py [hunk 1]
```diff
@@ -1197,14 +1189,53 @@ def aks_kollect(cmd,    # pylint: disable=too-many-statements,too-many-locals
...
 
-    return data_loaded
+    # Create YAML list items for each config variable that has a value
+    diag_content = "\n".join(f'  - {k}="{v}"' for k, v in diag_config_vars.items() if v is not None)
+
+    # Build a Kustomize overlay referencing a base for a known release, and using the images from MCR
+    # for that release.
+    return f"""
+resources:
+- https://github.com/{CONST_PERISCOPE_REPO_ORG}/aks-periscope//deployment/base?ref={CONST_PERISCOPE_RELEASE_TAG}
```
[the remark this answers is not in the document]
Comment (peterbom) [reply src/aks-preview/azext_aks_preview/custom.py:1213]: That's [correct](https://github.com/kubernetes-sigs/kustomize/blob/master/examples/remoteBuild.md#remote-directories).

> The directory is specified by appending a // after the repo URL

**Comment by peterbom** [comment]
> Yeah, would appreciate if you could add some unit test for these commands, since they are not related to ARM resource, I suppose live/e2e test based on vcr/recording is not suitable.

Discussed with @FumingZhang offline: we will work on one or two follow-up PRs to shuffle the code around (modelled on the recent `draft` additions) to make it more amenable to unit tests.

Further discussion may be required on whether the 'live' tests are necessary/appropriate.

**Review by zhoxing-ms — APPROVED** [review]

**Comment by FumingZhang** [comment]
The [CI](https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=56059212&view=results) failed due to changes in the behavior of the rp and will be fixed in another [PR #4919](https://github.com/Azure/azure-cli-extensions/pull/4919).

Queued a [pipeline](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=56060065&view=results) to test the newly added test case.

**Pull request merged**